src/lib.rs

```diff
@@ -187,6 +187,25 @@ fn concat<T>(mut v1: Vec<T>, mut v2: Vec<T>) -> Vec<T> {
     v1
 }
 
+fn check_for_default(triples: &mut Vec<(TokenStream, Ident, TokenTree)>) {
+    let mut default_position: Option<usize> = None;
+    for (attributes, variant_name, variant_value) in triples.into_iter() {
+        if attributes.to_string().contains("default") {
+            if default_position.is_some() {
+                // error!("Multiple variants marked as default");
```

Comment:
Did you mean to get rid of this if statement? I'm not 100% sure, but I would assume that `derive(Default)` itself will trigger an error if it sees multiple defaults.

Also it looks like we don't actually need to do anything with the default variant, so you probably don't need the `default_position` variable.

Reply:
So, um, it kinda triggers a [error](https://github.com/rust-lang/rust/issues/118119#issue-2003424948)

Reply:
Got it. Looks like someone opened a pr to fix the error message [here](https://github.com/rust-lang/rust/pull/118131) in response to the issue you opened.

Still, I suppose I'd prefer it if we propagated the error, at least until that other pr lands. Yea, as you probably noticed, we can't just use the `error!` macro there because your function doesn't return a TokenStream. I think the "correct" thing to do here would be to return `Result<(), String>` from `check_for_default` and check for an error when you call it, then call `error!` from there.